meson.build

```diff
@@ -0,0 +1,195 @@
+project('fluxengine', 'cpp', 'c',
+  default_options: ['cpp_std=c++2a']
+)
+add_global_arguments('-fext-numeric-literals', language: 'cpp')
+add_global_arguments('-Wno-deprecated-enum-enum-conversion', language: 'cpp')
+add_global_arguments('-Wno-deprecated-enum-float-conversion', language: 'cpp')
+
+zlib_dep = dependency('zlib')
+protobuf_dep = dependency('protobuf')
+pthreads_dep = dependency('threads')
+protoc = find_program('protoc', required : true)
+
+protoc_generator = generator(
+  protoc,
+  output: ['@BASENAME@.pb.cc', '@BASENAME@.pb.h'],
+  arguments: ['--proto_path=@CURRENT_SOURCE_DIR@', '--cpp_out=@BUILD_DIR@', '@INPUT@'],
+)
+
+subdir('dep/agg')
+subdir('dep/fmt')
+subdir('dep/libusbp')
+subdir('dep/stb')
```

Comment:
In general, you probably want to use https://mesonbuild.com/Wrap-dependency-system-manual.html here.

You can move dependencies to subprojects/ and write isolated meson.build files for them which export a sandboxed dependency interface and serve as a convention for including external dependencies.

Also for fmt specifically, you can just import it via `meson wrap install fmt`, switch to looking it up via `dependency()`, and seamlessly support debundled copies of the fmt library which may be available as part of the system on Linux.